libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -9,21 +9,36 @@ import {
   BooksPartialState
 } from '@tmo/books/data-access';
 import { FormBuilder } from '@angular/forms';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
 import { Book, DefaultTerm } from '@tmo/shared/models';
+import { Subject } from 'rxjs';
 
 @Component({
   selector: 'tmo-book-search',
   templateUrl: './book-search.component.html',
   styleUrls: ['./book-search.component.scss']
 })
-export class BookSearchComponent {
+export class BookSearchComponent implements OnInit, OnDestroy{
   readingListBook$ = this.store.select(getAllBooks);
   bookFetchError$ = this.store.select(getBooksError);
+  private destroyed$: Subject<boolean> = new Subject();
 
   searchForm = this.fb.group({
     term: ''
   });
 
+  ngOnInit(): void {
+    this.searchForm.get('term').valueChanges
+      .pipe(
+        takeUntil(this.destroyed$),
```

Comment:
It is better to use manual `subscription.unsubscribe()` since the `takeUntil` pattern can lead to problems if used in the wrong position.